### PR TITLE
DATAUP-330: util base dir test fixes

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Minor fixes to tests and modules in the `util/` directory.

Turn on Karma's setting to ensure that tests fail if there are no expectations.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-330
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and running the tests

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/tour.js [hunk 1]
```diff
@@ -46,10 +46,10 @@ define([
                 placement: 'bottom',
                 content:
                     "If there's an update to the Narrative Interface, this button will appear. Click it before you start working on your Narrative.",
-                onShow: function (tour) {
+                onShow: function () {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/tour.js:49]: minor eslint fixes and removal of comments in this file

File: kbase-extension/static/kbase/js/userMenu.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ define([
     'use strict';
 
     function factory(config) {
-        let target = config.target,
+        const target = config.target,
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/userMenu.js:13]: minor eslint fix

File: test/unit/karma.conf.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ module.exports = function (config) {
             jasmine: {
                 failFast: false,
                 DEFAULT_TIMEOUT_INTERVAL: 20000,
+                failSpecWithNoExpectations: true,
```
Comment (ialarmedalien) [inline test/unit/karma.conf.js:11]: Don't let tests pass if they aren't testing anything!

File: test/unit/spec/util/BootstrapDialogSpec.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 define(['jquery', 'util/bootstrapDialog'], ($, Dialog) => {
     'use strict';
     const $simpleBody = $('<div>').append('This is a body text'),
-        simpleTitle = 'Title',
+        simpleTitle = 'Simple Modal Title',
```
Comment (ialarmedalien) [inline test/unit/spec/util/BootstrapDialogSpec.js:4]: Changed the title to make the source of the modal easier to find in the test specs.

**Review by briehl — APPROVED** [review]
Great.

File: test/unit/spec/util/BootstrapDialogSpec.js [hunk 5]
```diff
@@ -10,27 +10,38 @@ define(['jquery', 'util/bootstrapDialog'], ($, Dialog) => {
...
+    }
 
     describe('Test the BootstrapDialog module', () => {
+        let simpleDialog;
+        afterEach(() => {
+            try {
+                simpleDialog.destroy();
+                // eslint-disable-next-line no-empty
+            } catch (err) {}
+            simpleDialog = null;
```
Comment (ialarmedalien) [inline test/unit/spec/util/BootstrapDialogSpec.js:29]: clean up afterwards

File: test/unit/spec/util/BootstrapSearchSpec.js [hunk 6]
```diff
@@ -91,8 +91,10 @@ define(['jquery', 'util/bootstrapSearch', 'base/js/namespace'], ($, BootstrapSea
         it('Should have a working focus function', (done) => {
             const bsSearch = new BootstrapSearch($targetElem);
             $('body').append($targetElem);
-            $targetElem.find('input.form-control').on('focus', (event) => {
+            spyOn(Jupyter.narrative, 'disableKeyboardManager');
+            $targetElem.find('input[type="text"]').on('focus', (event) => {
                 expect(event.type).toBe('focus');
+                expect(Jupyter.narrative.disableKeyboardManager).toHaveBeenCalled();
```
Comment (ialarmedalien) [inline test/unit/spec/util/BootstrapSearchSpec.js:97]: This test is super flaky -- not sure why. It frequently times out for no reason that I can see.

File: test/unit/spec/util/DisplaySpec.js [hunk 7]
```diff
@@ -9,6 +9,7 @@ define(['jquery', 'util/display', 'narrativeConfig', 'util/string'], (
     describe('KBase Display Utility function module', () => {
         let $nameTarget;
         const profilePageUrl = Config.url('profile_page');
+        const iconUrl = 'http://localhost:9876/narrative/static/kbase/images/ajax-loader.gif';
```
Comment (ialarmedalien) [inline test/unit/spec/util/DisplaySpec.js:12]: use a local link instead of fetching an icon from the kbase servers

**Pull request merged**